lib/cutting_edge/app.rb

```diff
@@ -62,17 +62,30 @@ class App < Sinatra::Base
     set :views, ::File.join(::File.dirname(__FILE__), 'templates')
     Tilt.register Tilt::ERBTemplate, 'html.erb'
     set :public_folder, ::File.join(::File.dirname(__FILE__), 'public')
-
     logger filename: "#{settings.environment}.log", level: :trace
 
     before do
       @store = settings.store
     end
 
     get '/' do
-      @repos = CuttingEdge::App.repositories.select {|_, repo| !repo.hidden?}
+      hidden_repos, public_repos = CuttingEdge::App.repositories.partition{|_, repo| repo.hidden?}.map(&:to_h)
+      @hidden_repos_exist = !hidden_repos.empty?
+      @repos = public_repos
       erb :index
     end
+    
+    post '/' do
+      payload = JSON.parse(request.body.read)
+      if defined?(::CuttingEdge::SECRET_TOKEN) && payload['token'] == ::CuttingEdge::SECRET_TOKEN
+        @repos = CuttingEdge::App.repositories.select{|_, repo| repo.hidden?}
```

Comment:
I really like this approach, but maybe we can make more clear what's happening by renaming this route to `/hidden_repos`?

Reply:
Yup.

Reply:
Done.